File: JS/billDivision.js
function bonAppetit(bill, k, b) {
    
    bill.splice(k, 1)
    
    const sharedItems = bill.reduce((acc, curr) => {
        return acc+curr
    })

    sharedItems/2===b ? console.log('Bon Appetit') : console.log(b-(sharedItems/2))
}

bonAppetit([3, 10, 2, 9], 1, 12)

File: Python/bill_division.py
'''
    bonAppetit has the following parameter(s):
        bill: an array of integers representing the cost of each item ordered
        k: an integer representing the zero-based index of the item Anna doesn't eat
        b: the amount of money that Anna contributed to the bill
    If Brian did not overcharge Anna, print Bon Appetit on a new line; otherwise, print the difference that Brian must refund to Anna. This will always be an integer.
'''
import functools

def bon_appetit(bill, k, b):
    value_total = functools.reduce(lambda acc, cost_item: acc + cost_item, bill)
    value_anna = (value_total - bill[k])/2
    
    print('Bon Appetit' if value_anna >= b else int(b - value_anna))

#example
bon_appetit([3, 10, 2, 9], 1, 12)
